Source/WebCore/platform/network/HTTPHeaderMap.h

```diff
@@ -178,10 +178,10 @@ class HTTPHeaderMap {
     WEBCORE_EXPORT bool remove(HTTPHeaderName);
 
     // Instead of passing a string literal to any of these functions, just use a HTTPHeaderName instead.
-    template<size_t length> String get(const char (&)[length]) const = delete;
-    template<size_t length> void set(const char (&)[length], const String&) = delete;
-    template<size_t length> bool contains(const char (&)[length]) = delete;
-    template<size_t length> bool remove(const char (&)[length]) = delete;
+    template<size_t length> String get(ASCIILiteral) const = delete;
+    template<size_t length> void set(ASCIILiteral, const String&) = delete;
+    template<size_t length> bool contains(ASCIILiteral) = delete;
+    template<size_t length> bool remove(ASCIILiteral) = delete;
```

Comment:
Are these still helpful? I assume they are, but I am not sure.

Reply:
Yes, they are still helpful because there is an overload that takes in a String and we wouldn't want someone calling the String overload using an ASCIILiteral (which would get implicitly converted).

Instead of calling these functions with an ASCIILiteral, the caller is supposed to pass a HTTPHeaderName enum value.

Reply:
Would be more elegant if those overloads took a `StringView`. But I suppose we would still need to delete `ASCIILiteral `overloads in that case too for the same reason.